src/Notifier.cpp

```diff
@@ -96,9 +119,36 @@ size_t Notifier::Notify() {
       LOG(WARNING) << "Uninitialized notifier queues";
       return {0};
    }
-   const std::string message = "notify";
-   LOG(INFO) << message;
-   gQueue->Fire(message);
+   LOG(INFO) << "Notifier: Sending notification message: " << notifyMessage;
+   gQueue->Fire(notifyMessage);
+   size_t confirmed = ReceiveConfirmation();
+   LOG(INFO) << "Notifier received " << confirmed << " handshakes";
+   return {confirmed};
+}
+
+/*
+ * Fire a message from the Shotgun to be
+ *    read by the queue subscriber. Then wait for listeners
+ *    to confirm they received the notification
+ * 
+ * @param vector of strings to be sent to the listeners             
+ * @return number of confirmed updates
+ */
+size_t Notifier::Notify(const std::vector<std::string>& messages) {
+   std::lock_guard<std::mutex> guard(gLock);
+   std::vector<std::string> bullets;
+   bullets.push_back(notifyMessage);
+
+   for (auto& msg : messages) {
+      bullets.push_back(msg);
+   }
+
+   if (QueuesAreUnitialized()) {
+      LOG(WARNING) << "Uninitialized notifier queues";
```

Comment:
I guess warning is OK. Can you see what the other patterns are using? If they use FATAL here, then please use FATAL